nerdctl panics on missing XDG_RUNTIME_DIR env var

Expected behavior: I'd expect it to exit more cleanly where XDG_RUNTIME_DIR is really needed & continue on where it isn't needed

```bash
$ nerdctl completion bash

panic: environment variable XDG_RUNTIME_DIR is not set

goroutine 1 [running]:
github.com/AkihiroSuda/nerdctl/pkg/defaults.BuildKitHost(0x203000, 0x203000)
        github.com/AkihiroSuda/nerdctl/pkg/defaults/defaults.go:86 +0x105
main.init()
        github.com/AkihiroSuda/nerdctl/build.go:42 +0x34
```

--- a/pkg/defaults/defaults.go
+++ b/pkg/defaults/defaults.go
@@ -24,6 +24,7 @@ import (
 
 	"github.com/AkihiroSuda/nerdctl/pkg/rootlessutil"
 	gocni "github.com/containerd/go-cni"
+	"github.com/sirupsen/logrus"
 )
 
 const AppArmorProfileName = "nerdctl-default"
@@ -83,7 +84,8 @@ func BuildKitHost() string {
 	}
 	xdr, err := rootlessutil.XDGRuntimeDir()
 	if err != nil {
-		panic(err)
+		logrus.Warn(err)
+		xdr = fmt.Sprintf("/run/user/%d", rootlessutil.ParentEUID())
 	}
 	return fmt.Sprintf("unix://%s/buildkit/buildkitd.sock", xdr)
 }

--- a/pkg/rootlessutil/rootlessutil.go
+++ b/pkg/rootlessutil/rootlessutil.go
@@ -17,6 +17,43 @@
 
 package rootlessutil
 
+import (
+	"os"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
 func IsRootless() bool {
 	return IsRootlessParent() || IsRootlessChild()
+}
+
+func ParentEUID() int {
+	if !IsRootlessChild() {
+		return os.Geteuid()
+	}
+	env := os.Getenv("ROOTLESSKIT_PARENT_EUID")
+	if env == "" {
+		panic("environment variable ROOTLESSKIT_PARENT_EUID is not set")
+	}
+	i, err := strconv.Atoi(env)
+	if err != nil {
+		panic(errors.Wrapf(err, "failed to parse ROOTLESSKIT_PARENT_EUID=%q", env))
+	}
+	return i
+}
+
+func ParentEGID() int {
+	if !IsRootlessChild() {
+		return os.Getegid()
+	}
+	env := os.Getenv("ROOTLESSKIT_PARENT_EGID")
+	if env == "" {
+		panic("environment variable ROOTLESSKIT_PARENT_EGID is not set")
+	}
+	i, err := strconv.Atoi(env)
+	if err != nil {
+		panic(errors.Wrapf(err, "failed to parse ROOTLESSKIT_PARENT_EGID=%q", env))
+	}
+	return i
 }

--- a/pkg/rootlessutil/xdg.go
+++ b/pkg/rootlessutil/xdg.go
@@ -33,7 +33,7 @@ func XDGRuntimeDir() (string, error) {
 	if euid := os.Getenv("ROOTLESSKIT_PARENT_EUID"); euid != "" {
 		return "/run/user/" + euid, nil
 	}
-	return "", errors.New("environment variable XDG_RUNTIME_DIR is not set")
+	return "", errors.New("environment variable XDG_RUNTIME_DIR is not set, see https://rootlesscontaine.rs/getting-started/common/login/")
 }
 
 func XDGConfigHome() (string, error) {

